zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -33,3 +51,694 @@ pub(crate) fn detect_spending_accounts<'a>(
+           ":block": block,
+           ":mined_height": u32::from(output.height())
+        ],
+        |row| row.get::<_, i64>(0),
+    )?;
+
+    let mut stmt_upsert_transparent_output = conn.prepare_cached(
+        "INSERT INTO transparent_received_outputs (
+            transaction_id, output_index,
+            account_id, address, script,
+            value_zat, max_observed_unspent_height
+        )
+        VALUES (
+            :transaction_id, :output_index,
+            :account_id, :address, :script,
+            :value_zat, :height
+        )
+        ON CONFLICT (transaction_id, output_index) DO UPDATE
+        SET account_id = :account_id,
+            address = :address,
+            script = :script,
+            value_zat = :value_zat,
+            max_observed_unspent_height = :height
+        RETURNING id",
+    )?;
+
+    let sql_args = named_params![
+        ":transaction_id": id_tx,
+        ":output_index": &output.outpoint().n(),
+        ":account_id": received_by_account.0,
+        ":address": &output.recipient_address().encode(params),
+        ":script": &output.txout().script_pubkey.0,
+        ":value_zat": &i64::from(Amount::from(output.txout().value)),
+        ":height": &u32::from(output.height()),
+    ];
+
+    stmt_upsert_transparent_output.query_row(sql_args, |row| row.get::<_, i64>(0).map(UtxoId))
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        testing::{AddressType, TestBuilder, TestState},
+        PRUNING_DEPTH,
+    };
+    use sapling::zip32::ExtendedSpendingKey;
+    use zcash_client_backend::{
+        data_api::{
+            wallet::input_selection::GreedyInputSelector, InputSource, WalletRead, WalletWrite,
+        },
+        encoding::AddressCodec,
+        fees::{fixed, DustOutputPolicy},
+        wallet::WalletTransparentOutput,
+    };
+    use zcash_primitives::{
+        block::BlockHash,
+        consensus::BlockHeight,
+        transaction::{
+            components::{amount::NonNegativeAmount, OutPoint, TxOut},
+            fees::fixed::FeeRule as FixedFeeRule,
+        },
+    };
+
+    #[test]
+    fn put_received_transparent_utxo() {
+        use crate::testing::TestBuilder;
+
+        let mut st = TestBuilder::new()
+            .with_account_from_sapling_activation(BlockHash([0; 32]))
+            .build();
+
+        let account_id = st.test_account().unwrap().account_id();
+        let uaddr = st
+            .wallet()
+            .get_current_address(account_id)
+            .unwrap()
+            .unwrap();
+        let taddr = uaddr.transparent().unwrap();
+
+        let height_1 = BlockHeight::from_u32(12345);
+        let bal_absent = st
+            .wallet()
+            .get_transparent_balances(account_id, height_1)
+            .unwrap();
+        assert!(bal_absent.is_empty());
+
+        // Create a fake transparent output.
+        let value = NonNegativeAmount::const_from_u64(100000);
+        let outpoint = OutPoint::new([1u8; 32], 1);
```

Comment:
The rebase of the first move-only commit undid the `OutPoint::fake()` helper introduction from #1423 (6d35583a4b7eff04c094a82a16be83a7e97727a1). Fix up the first move-only commit (and the second if necessary) to preserve the change.